services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/models/shared_store.py

```diff
@@ -14,6 +23,40 @@ class SharedStore(BaseModel):
         description="stores the container names from the compose_spec",
     )
 
-    def clear(self):
+    async def clear(self):
         self.compose_spec = None
         self.container_names = []
+        await self.persist_to_disk()
+
+    @classmethod
+    async def init_from_disk(cls, shared_store_dir: Path) -> "SharedStore":
+        data_file_path = shared_store_dir / STORE_FILE_NAME
+        if data_file_path.exists():
+            # if the sidecar is started for a second time (usually the container dies)
+            # it will load the previous data which was stored
+            async with aiofiles.open(shared_store_dir / STORE_FILE_NAME) as data_file:
+                file_content = await data_file.read()
+
+            obj = cls.parse_obj(file_content)
+        else:
+            obj = cls()
+
+        obj._shared_store_dir = shared_store_dir  # pylint: disable=protected-access
```

Comment:
as pylint suggests, this does not look like it should be.

Reply:
I disagree with pylint. It does not know the internals of pydantic. In this situation I required `_shared_store_dir` to be **"private"**.

Reply:
I've only managed to add a "set_..." method for this. Could not use python's setterrs and properties. Somehow pydantic interferes with it.